### PR TITLE
rewire aggregation handling chain to be normal

### PR DESCRIPTION
**Pull request by deads2k**
Uses https://github.com/kubernetes/kubernetes/pull/42886 to allow the aggregator to be "normal" as far as the handling chain goes.  This will allow for cleaner composition.

@kubernetes/sig-api-machinery-misc

### CODE REVIEW

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -165,43 +157,6 @@ func (c completedConfig) New(stopCh <-chan struct{}) (*APIAggregator, error) {
...
-	handler = genericapifilters.WithAudit(handler, c.RequestContextMapper, os.Stdout)
-	handler = genericapifilters.WithAuthentication(handler, c.RequestContextMapper, c.Authenticator, genericapifilters.Unauthorized(c.SupportsBasicAuth))
-
-	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
-	handler = genericfilters.WithPanicRecovery(handler, c.RequestContextMapper)
-	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.RequestContextMapper, c.LongRunningFunc)
-	handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.RequestContextMapper, c.LongRunningFunc)
-	handler = genericapifilters.WithRequestInfo(handler, genericapiserver.NewRequestInfoResolver(c), c.RequestContextMapper)
-	handler = genericapirequest.WithRequestContext(handler, c.RequestContextMapper)
-
```
Comment (sttts) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go]: 👍

File: staging/src/k8s.io/apiserver/pkg/server/mux/container.go [hunk 2]
```diff
@@ -35,22 +35,16 @@ import (
...
 // NewAPIContainer constructs a new container for APIs
-func NewAPIContainer(mux *http.ServeMux, s runtime.NegotiatedSerializer) *APIContainer {
+func NewAPIContainer(mux *http.ServeMux, s runtime.NegotiatedSerializer, defaultMux http.Handler) *APIContainer {
 	c := APIContainer{
 		Container: restful.NewContainer(),
-		NonSwaggerRoutes: PathRecorderMux{
-			mux: mux,
-		},
-		UnlistedRoutes: mux,
+		// NonSwaggerRoutes: PathRecorderMux{
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/mux/container.go]: remove the old code

**Comment by sttts** [comment]
Looks good.

**Comment by deads2k** [comment]
@k8s-bot non-cri e2e test this

**Comment by deads2k** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**